test/src/Queue/SqsWatchableQueueTest.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+namespace test\eLife\Bus\Queue;
+
+use Aws\Command;
+use Aws\MockHandler;
+use Aws\Result;
+use Aws\Sqs\SqsClient;
+use eLife\Bus\Queue\BusSqsMessage;
+use eLife\Bus\Queue\SqsWatchableQueue;
+use PHPUnit\Framework\TestCase;
+use Traversable;
+
+final class SqsWatchableQueueTest extends TestCase
+{
+    /** @var SqsClient */
+    private $sqsClient;
+    /** @var MockHandler */
+    private $mockHandler;
+
+    /**
+     * @before
+     */
+    public function setUpSqsClient()
+    {
+        $this->sqsClient = new SqsClient(['region' => 'us-east-1', 'version' => '2012-11-05']);
+        $this->mockHandler = new MockHandler();
+
+        $this->sqsClient->getHandlerList()->setHandler($this->mockHandler);
+
+        $this->mockHandler->append(new Result(['QueueUrl' => 'http://www.example.com/']));
+    }
+
+    /**
+     * @test
+     */
+    public function it_adds_an_item_to_the_queue()
+    {
+        $queue = new SqsWatchableQueue($this->sqsClient, 'queue');
+
+        $result = [];
+        $this->mockHandler->append(function (Command $command) use (&$result) {
```

Comment:
I usually put collectors like `$result` on `$this` so that they are easier to pass around